src/java/com/palantir/cassandra/db/IColumnFamilyStoreValidator.java

```diff
@@ -33,4 +33,22 @@ public interface IColumnFamilyStoreValidator
      */
     Map<Descriptor, Set<Integer>> filterValidAncestors(CFMetaData cfMetaData,
                                                        Map<Descriptor, Set<Integer>> sstableToCompletedAncestors, Map<Integer, UUID> unfinishedCompactions);
+
+    /**
+     * @return true if Cassandra should use ancestry metdata to cleanup unused SSTables on startup by running
+     * {@link org.apache.cassandra.db.ColumnFamilyStore#removeUnusedSstables(CFMetaData, Map)}, false otherwise
+     * (e.g. if a different cleanup system is being used outside of {@link org.apache.cassandra.service.CassandraDaemon}).
+     */
+    default boolean shouldRemoveUnusedSstables() {
```

Comment:
nit: can the method name include some reference to this being for the native/default/ancestor-derived behavior, instead of just in the javadoc?

Reply:
renamed to `shouldSkipAncestorCleanupBasedOnAncestorMetadata`